Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -790,43 +790,60 @@ namespace OpenInfraPlatform {
 					}
 				}
 				
+				/*! \brief  Converts a list of \c IfcFaceBound -s to a list of boundary loops.
+				 * The first loop in the return list is converted from the \c IfcFaceOuterBound entity.
+				 * Only one boundary should be of this type.
+				 *
+				 * \param[in]	ifcFaceBounds	List of \c IfcFaceBound entities to be converted.
+				 * \param[in]	pos				The relative location of the origin of the representation's coordinate system within the geometric context.
+				 * \return		List of boundary loops.
+				 */
 				std::vector<std::vector<carve::geom::vector<3>>> convertIfcFaceBoundList(
-					const std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcFaceBound>>& ifcFaceBounds,
+					std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcFaceBound>> ifcFaceBounds,
 					const carve::math::Matrix& pos) const noexcept(true)
 				{
-					const std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcFaceBound>> faceBounds = swapOuterBoundaryToFront(ifcFaceBounds);
+					swapOuterBoundaryToFront(ifcFaceBounds);
 
 					std::vector<std::vector<carve::geom::vector<3>>> faceBoundLoops;
-					faceBoundLoops.resize(faceBounds.size());
-					for (size_t i = 0; i < faceBounds.size(); i++)
-					{
-						// get vertices of one bound loop into faceBoundLoops[i]
-						faceBoundLoops[i] = convertIfcFaceBound(faceBounds[i], pos);
-					}
+					faceBoundLoops.resize(ifcFaceBounds.size());
+
+					// for each bound loop: get vertices into faceBoundLoops[i]
+					std::transform(ifcFaceBounds.begin(), ifcFaceBounds.end(), faceBoundLoops.begin(),
+						[this, &pos](EXPRESSReference<typename IfcEntityTypesT::IfcFaceBound> ifcFaceBound) { return convertIfcFaceBound(ifcFaceBound, pos); });
 
 					return faceBoundLoops;
 				}
 
-				std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcFaceBound>> swapOuterBoundaryToFront(
-					std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcFaceBound>> ifcFaceBounds) const throw(...)
+				/*! \brief  Swaps the \c IfcFaceOuterBound entity to the first position in the list of \c IfcFaceBound entities.
+				 * Only one boundary loop can be an \c IfcFaceOuterBound entity. But in general, it's optional. 
+				 * \param[in,out]	ifcFaceBounds	List of \c IfcFaceBound entities in which the \c IfcFaceOuterBound entity has to be swaped to the front.
+				 */
+				void swapOuterBoundaryToFront(
+					std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcFaceBound>>& ifcFaceBounds
+				) const throw(...)

```

Comment:
suggestion:

throw if there are more than 1 outer boundaries - you can use `std::count_if`

Reply:
`throw` is introduced with commit a9990e7.
I'm not sure about the throw message: Usually, the custom `throw`s expect a single `EXPRESSReference` entity (if I understood it right). However, at this point the function has a bunch (vector) of entities which causes the throw. Just to give the user a hint about from where the error comes, I put the id of at least one entity into the string-message overload.
When trace back to a single source entity, the `IfcFace` in the function `convertIfcFace` would be the appropriate one - but currently, the function doesn't have access to it. To add it as a parameter just for the throw feels like not the best choice.

Reply:
Good analysis. No need for a special parameter for a `throw`. Your solution seems fine.